martian/src/metadata.rs

```diff
@@ -146,64 +153,69 @@ impl Metadata {
         let metadata_path = args.pop().unwrap();
         let stage_type = args.pop().unwrap();
         let stage_name = args.pop().unwrap();
-        let alarm_file = SharedFile::new(make_metadata_file_path(metadata_path.as_ref(), "alarm"));
 
-        Metadata {
+        Ok(Metadata {
             stage_name,
-            stage_type,
+            stage_type: stage_type.parse()?,
             metadata_path,
             files_path,
             run_file,
             raw_jobinfo: Map::new(),
             jobinfo: Default::default(),
-            alarm_file,
-        }
+        })
     }
 
     /// Path within chunk
     pub fn make_path(&self, name: &str) -> PathBuf {
         make_metadata_file_path(self.metadata_path.as_ref(), name)
     }
 
-    /// Write to a file inside the chunk
-    pub fn write_raw(&mut self, name: &str, text: &str) -> Result<()> {
-        let mut f = File::create(self.make_path(name))?;
-        f.write_all(text.as_bytes())?;
-        // Ensure the file is closed before we write the journal, to reduce
-        // the chances that `mrp` sees the journal entry before the file content
-        // has be sync'ed.  This can be an issue on nfs systems.
+    /// Append to a metadata file inside the chunk.
+    ///
+    /// This operation is not atomic at the file system level.
+    pub fn append_metadata(&mut self, name: &str, content: &str) -> Result<()> {
```

Comment:
could this just be `pub(crate)`? I'm hesitant to add additional public API surface area here, in part because doing so limits compiler optimizations.

Reply:
I kinda think that all of `Metadata` should probably be `pub(crate)`; I think the only reason it isn't is because it appears in the signature of `RawMartianStage`. I'd be fine with making this API pub(crate).